kafka-client/src/main/java/dev/responsive/kafka/config/ResponsiveStreamsConfig.java

```diff
@@ -0,0 +1,65 @@
+package dev.responsive.kafka.config;
+
+import static dev.responsive.kafka.api.SharedClients.sharedClientConfigs;
+
+import dev.responsive.db.CassandraClient;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.ScheduledExecutorService;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.streams.StreamsConfig;
+import org.apache.kafka.streams.StreamsConfig.InternalConfig;
+import org.apache.kafka.streams.processor.internals.assignment.StickyTaskAssignor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ResponsiveStreamsConfig {
```

Comment:
yeah, that's fair. At first I thought I wanted to have it extend StreamsConfig but I ended up backing that out, so it does feel a bit misleading right now. I'll make those changes

Reply:
Actually, I'm just going to move this into `ResponsiveKafkaStreams`. I think it should be semi-visible to users because it's where we define the config overrides that users will need to set

Btw I still think it's super awkward to ask users to set the internal task assignor config...it's not a real config. Would you be willing accept us overriding this config if unset, and only throwing an exception if they had tried to set it to something else already? It hardly makes sense right now since users either don't care or would be setting it to the StickyTaskAssignor already, which is the only other real assignor. In the very very rare case they plug in a custom assignor of their own, we of course throw an exception.

In the future, we'll have to ask them to change this when we move to using our own custom assignor, which is annoying. At that point, we would throw an exception if the user set it to anything themselves, sticky task assignor included.

Won't push too hard on this, just feels ridiculous to ask them to set an internal config while knowing we're going to have to ask them to change this again probably not long from now. (Or we can just copy the sticky task assignor code into a custom assignor class and have them plug that in right away?)

Reply:
Oh I'm totally Ok (in fact I think we should be) overriding defaults! I just think we should throw an exception in the case where for some reason they did explicitly set it.